app/plugin.go

```diff
@@ -429,11 +431,29 @@ func (a *App) ShutDownPlugins() {
 	a.PluginEnv = nil
 }
 
+func getKeyHash(pluginId, key string) (string, *model.AppError) {
+	if len(pluginId) == 0 {
+		return "", model.NewAppError("getKeyHash", "app.plugin.key_value.plugin_id.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	if len(key) == 0 {
+		return "", model.NewAppError("getKeyHash", "app.plugin.key_value.key.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	hash := fnv.New128a()
+	hash.Write([]byte(pluginId + key))
+	return base64.StdEncoding.EncodeToString(hash.Sum(nil)), nil
```

Comment:
Went with fnv because I didn't think it needed to be cryptographic and it was faster, but the plugin creator does have control over what the plugin ID can be so they might be able to craft collisions (?)

Thoughts? Maybe going with sha is better despite being slower

Reply:
I'm leaning towards SHA. Also thinking a binary column type so we can save space / time by not base64 or hex encoding it.